### ISSUE TITLE
[DOC] Please provide a doc how-to integrate k0s

### ISSUE THREAD

**Issue by vanvuongngo** [harvester/harvester#6834]
**Is your doc request related to a problem? Please describe or add the related issue ID.**

Hi, I got stuck with get started with Harvester.
Set up Harvester itself is quite easy.

But when it comes to setup a K8S Cluster its difficult.
Three approach I testet...
1. use the embedded Rancher, works until I upgraded Harvester then my cluster got unuseable and its also not recommended for production
2. Setup a separated HA k3s Cluster only for Rancher, but thats difficult to setup with the certs w/o an expose to the Internet. And this approach requires too much resources and causes extra maintain for the edge
3. Use Harvester to create VMs and set them up with k0s HA cluster is simple with the k0sctl CLI

With the third approach I got an k8s running but when I try to install sth. with Helm and claim for persistent volume it keeps on stuck in pending state.

How can I connect k0s with Harvester?

**Describe the solution you'd like**

Please provide a howto with k0s to lower entry level into Harvester.

**Comment by w13915984028** [comment]
@vanvuongngo You may refer https://docs.harvesterhci.io/v1.3/rancher/csi-driver. When your guest cluster is running workload which claims for persistant volume, then the guest cluster needs kind of CSI driver to support it. 

Harvester has CSI-dirver, which is well tested for integration with Rancher Manager & RKE2. Hopefully you will get it also working on k0s, thanks.

**Comment by mcuw** [comment]
Hi @w13915984028, thanks a lot for your quick solution and I will try it out :)

**Comment by vanvuongngo** [comment]
Hi, I closed this topic because I found [Talos Linux](https://talos.dev) which is inherently more secure and has less maintenance than k3s and k0s. Since there is already a CSI plugin for proxmox, I will probably switch back to proxmox.  Which I find very sad, as I think harvester is quite promising. Only the documentation and support of other solutions beyond the Suse solutions to Rancher and k3s is expandable. If there is an extended documentation for CSI, I would be very grateful.
